apisix/init.lua

```diff
@@ -630,10 +630,11 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
-    local status_code = tonumber(get_var("upstream_status"))
-    if status_code and status_code >= 500 and status_code <= 599 then
-        core.response.set_header("X-APISIX-Upstream-Status", status_code)
-        core.log.info("X-APISIX-Upstream-Status: ", status_code)
+    local up_status = get_var("upstream_status")
+    local from, _ = re_find(up_status, "5[0-9]{2}", "jo")
+    if from then
+        core.response.set_header("X-APISIX-Upstream-Status", up_status)
```

Comment:
We should find a place to document this feature.

Reply:
done.